sql/src/main/java/io/crate/analyze/CreateTableStatementAnalyzer.java

```diff
@@ -94,4 +109,39 @@ public AnalyzedCreateTable analyze(CreateTable<Expression> createTable,
             analyzedTableElementsWithExpressions
         );
     }
+
+    private static List<TableElement<Symbol>> analyzeButCheckConstraints(CreateTable<Expression> createTable,
+                                                                         ExpressionAnalyzer exprAnalyzerWithFieldsAsString,
+                                                                         ExpressionAnalysisContext exprCtx) {
+        List<TableElement<Expression>> notCheckConstraints = createTable.tableElements()
+            .stream()
+            .filter(x -> false == x instanceof CheckConstraint)
+            .collect(Collectors.toList());
+        List<TableElement<Symbol>> analyzed = new ArrayList<>(notCheckConstraints.size());
+        Set<CheckColumnConstraint<Expression>> checkColumnConstraints = new HashSet<>(notCheckConstraints.size());
+        for (int i = 0; i < notCheckConstraints.size(); i++) {
+            TableElement<Expression> te = notCheckConstraints.get(i);
+            if (te instanceof ColumnDefinition) {
+                ColumnDefinition<Expression> def = (ColumnDefinition<Expression>) te;
+                List<ColumnConstraint<Expression>> constraints = def.constraints();
+                for (int j = 0; j < constraints.size(); j++) {
+                    ColumnConstraint<Expression> cc = constraints.get(j);
+                    if (cc instanceof CheckColumnConstraint) {
+                        CheckColumnConstraint<Expression> check = (CheckColumnConstraint<Expression>) cc;
+                        checkColumnConstraints.add(check);
+                        // Re-frame the column constraint as a table constraint
+                        createTable.tableElements().add(new CheckConstraint<>(
+                            check.name(),
+                            def.ident(),
+                            check.expression(),
+                            check.expressionStr()
+                        ));
+                    }
+                }
+                def.constraints().removeAll(checkColumnConstraints);
```

Comment:
why is this needed? if needed, could this maybe solved without mutating?

Reply:
In this case, what we are doing is treating a check constraint definition at the column level as if it were defined at the table level. During the first pass of the analysis we cannot process check constraints because we do not have references to the columns yet (they are being created).

Reply:
will have a better look at this